runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/logging/JulHandlerPrintStreamAdapterFactory.java

```diff
@@ -130,46 +128,50 @@ public synchronized void write(int i) {
     @Override
     public void write(byte[] a, int offset, int length) {
       ByteBuffer incoming = ByteBuffer.wrap(a, offset, length);
+      assert incoming.hasArray();
```

Comment:
could return early for length == 0, not obvious if this would hold in that case

Reply:
it still does hold, added code to return early for length == 0